File: Plus One.js
// https://leetcode.com/problems/plus-one/description/
// 66. Plus One

/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {

    let carry = true;
    let idx = digits.length - 1;

    while (carry){
        const sum = digits[idx] + 1;

        if(sum < 10) {
            digits[idx] = sum;
            carry = false;
        }
        else{
            digits[idx] = 0;
            if(idx === 0 ) {
                digits.unshift(1);
                carry = false;
            }
        }
        idx--;
    }

    return digits

};

File: Plus One.py
# https://leetcode.com/problems/plus-one/?envType=daily-question&envId=2026-01-01
# Plus One

class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        s = ""

        for digit in digits:
            s += str(digit)

        s = int(s)
        s += 1
        s = str(s)

        return [int(n) for n in s]